storage/google/cloud/storage/_helpers.py

```diff
@@ -33,7 +34,12 @@ class _PropertyMixin(object):
     """
 
     def __init__(self, name=None):
-        self.name = name
+        if name is None or (re.match(r'\w', name[0]) and
```

Comment:
I'm 100% open to alternative methods of checking each end of the name.
My regex foo is weak.